src/puppeteer-tests/utils/UtilityFunctions.js

```diff
@@ -0,0 +1,269 @@
+		}
+	}).then(res => {
+		const hit = res.cargo.hits.filter(thing => thing.name === name);
+		if (hit && hit[0] && hit[0].id) return hit[0].id;
+		return '';
+	});
+}
+
+// Goes to the given URL (which must contain a Good-loop ad), watches the video, and makes a donation 
+/**
+ * Advert must already be somewhere on the page before this method is called
+ * @param { object } page puppeteer test object
+ * @param { string } type(optional) behaviour needs to be slightly different for type:banner ads
+ */
+async function watchAdvertAndDonate({page, type}) {
+	await page.waitFor(1000);//Allow 'visible' event to register. Doesn't get counted if you start working right away
+	let pageOrIFrame = page; // If unit is wrapped in iframe, need to use iframe.ACTION instead of page.ACTION
+
+	// Adunit may have been loaded in to an iframe.
+	// Puppeteer will not cycle through frames to look for a given selector, so need to tell it where to look
+	// TODO cut down on possible values after this has been harmonised across the different pages/services
+	const iframe = await page.frames().find(f => f.name().slice(0, 2) === 'gl' || f.name() === 'test01' || f.name() === 'demo-iframe');
+	
+	if ( iframe ) {
+		pageOrIFrame = iframe;
+	}
+
+	// Banner-types
+	// Ad on Vpaid page should just autoplay
+	if( type === 'banner' ) {
+		await pageOrIFrame.waitForSelector(TestAs.Banner);
+		await pageOrIFrame.click(TestAs.Banner);            
+	}
+	try{
+		// Click-to-play video
+		await pageOrIFrame.waitForSelector(TestAs.ClickToPlay, { timeout: 3000 });
+		await pageOrIFrame.click(TestAs.ClickToPlay);
+	} catch (e) { 
+		console.log("Non click-to-play video. Continuing...");
+	}
+	// Pick a charity
+	await pageOrIFrame.waitForSelector('.chooser-list.ready');
+	await pageOrIFrame.click('a.charity');
+
+	await page.waitFor(5000);//Generally needs a second to register that donation has been made
+}
+
+// Filters through prop controls on the Studio page
+/**
+ * Studio page must be loaded before this is called
+ * @param { string } filter what to search for
+ * @param { string } name specify a custom name to confirm the search with - overrides filter
+ * @param { string } confirmSelector specify a custom selector to confirm the search with - overrides name and filter
+ * @param { object } Selectors reserve of common GL selectors to use
+ */
+async function filterProps ({filter, name, confirmSelector, Selectors=CommonSelectors}) {
```

Comment:
Let's have a file for studio utility functions - and keep this file for functions which apply to at least a couple of projects.

Reply:
Will do